### PR TITLE
Initial Implementation

### PR DESCRIPTION
**Pull request by Uzlopak**
This is the first implementation of the swagger-ui plugin. 

Only options which effect the exposeRoute and the swagger-ui from @fastify/swagger should be covered by this plugin. exposeRoute as an option was removed, as it is implied that we want to expose the routes if we use this plugin. 

Let us talk about if the initial options and separation is fine. 

Still have to fix some tests, but it should already be self explainatory. 

So please let us discuss if it is meeting our needs.

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/routes.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  }
+
+  return redirectPath
+}
+
+function fastifySwagger (fastify, opts, done) {
+  let staticCSP = false
+  if (opts.staticCSP === true) {
+    const csp = require('../dist/csp.json')
+    staticCSP = `default-src 'self'; base-uri 'self'; block-all-mixed-content; font-src 'self' https: data:; frame-ancestors 'self'; img-src 'self' data: validator.swagger.io; object-src 'none'; script-src 'self' ${csp.script.join(' ')}; script-src-attr 'none'; style-src 'self' https: ${csp.style.join(' ')}; upgrade-insecure-requests;`
```
Comment (Fdawgs) [inline lib/routes.js:26]:
```suggestion
    staticCSP = `default-src 'self'; base-uri 'self'; font-src 'self' https: data:; frame-ancestors 'self'; img-src 'self' data: validator.swagger.io; object-src 'none'; script-src 'self' ${csp.script.join(' ')}; script-src-attr 'none'; style-src 'self' https: ${csp.style.join(' ')}; upgrade-insecure-requests;`
```

`block-all-mixed-content` directive is deprecated, see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/block-all-mixed-content.

Can probably drop unless we're targeting IE... 5?

File: test/csp.test.js [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+  fastify.get('/parameters/:id', schemaParams, () => {})
+  fastify.get('/example1', schemaSecurity, () => {})
+
+  {
+    const res = await fastify.inject({
+      method: 'GET',
+      url: '/documentation/static/index.html'
+    })
+    t.equal(res.statusCode, 200)
+    t.equal(res.headers['content-security-policy'], `default-src 'self'; base-uri 'self'; block-all-mixed-content; font-src 'self' https: data:; frame-ancestors 'self'; img-src 'self' data: validator.swagger.io; object-src 'none'; script-src 'self' ${csp.script.join(' ')}; script-src-attr 'none'; style-src 'self' https: ${csp.style.join(' ')}; upgrade-insecure-requests;`)
```
Comment (Fdawgs) [inline test/csp.test.js:60]:
```suggestion
    t.equal(res.headers['content-security-policy'], `default-src 'self'; base-uri 'self'; font-src 'self' https: data:; frame-ancestors 'self'; img-src 'self' data: validator.swagger.io; object-src 'none'; script-src 'self' ${csp.script.join(' ')}; script-src-attr 'none'; style-src 'self' https: ${csp.style.join(' ')}; upgrade-insecure-requests;`)
```

**Review by Fdawgs — COMMENTED** [review]
Can get rid of `.npmignore` if its an exact copy of `.gitignore`, as NPM will fall back to using `.gitignore` if no `.npmignore` exists. See https://docs.npmjs.com/cli/v8/using-npm/developers#keeping-files-out-of-your-package

**Comment by Uzlopak** [comment]
removed the .npmignore

I still have to reactivate some unit tests and then it should be good to go.

**Review by mcollina — APPROVED** [review]
lgtm

**Review by Fdawgs — APPROVED** [review]

**Comment by fmenis** [comment]
@Uzlopak I don't really agree with `exposeRoute as an option was removed, as it is implied that we want to expose the routes if we use this plugin.`
Imho, the `exposeRoute` configuration is very useful.

So, from version 8, if we don't want to expose the routes documentation in production, we must not register the plugin at all?
Before version 8, it was very easy `exposeRoute: process.env.NODE_ENV !== ENV.PRODUCTION`

**Comment by Eomm** [comment]
Note that you need to write:

```
if(process.env.NODE_ENV !== ENV.PRODUCTION){
  fastify.register(swaggerUi)
}
```

What would be a better option?

**Comment by fmenis** [comment]
@Eomm for me, the best option would be the one exposed by the exposeRoute flag.
I find it more elegant than dynamically registering the plugin.
But it doesn't matter, as you (and I in the first post) indicated, just register the plugin dynamically and all works fine.

**Pull request merged**